### PR TITLE
boost: fix small things

### PR DESCRIPTION
**Pull request by mbodmer**
Specify library name and version:  **boost/1.xx.x**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by mbodmer** [comment]
Hi, I was looking at `recipes/boost/all/test_package/CMakeLists.txt`. Maybe it's a good Idea to use imported targets with TARGET_LINK_LIBRARIES.

So instead of eg. TARGET_LINK_LIBRARIES(random_exe ${Boost_LIBRARIES}) I would rather see TARGET_LINK_LIBRARIES(random_exe Boost::random)

This would make it easier to recognize mistakes such as forgotten required components, as it would not build. The cmake minimal version would have to be increased to 3.5, however. https://cmake.org/cmake/help/v3.5/module/FindBoost.html

**Comment by mbodmer** [comment]
If it would be appreciated, I could add these changes (imported targets in CMakeLists) to the PR also.

**Comment by danimtb** [comment]
@mbodmer go ahead with the changes! 👍

**Comment by mbodmer** [comment]
Could the build problem be related to the cmake version?

**Comment by danimtb** [comment]
this?
```
CMake Error at CMakeLists.txt:50 (ADD_EXECUTABLE):
  The target name "test" is reserved when CTest testing is enabled.
```
I have no idea but I'd rather change the name before to try to fix it if that's the case

File: recipes/boost/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PROJECT(MyHello)
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
```
Comment (uilianries) [inline recipes/boost/all/test_package/CMakeLists.txt]: Why do you need targets? Believe or not, cmake 2.8 is largely used yet, this change can break users.

Comment (mbodmer) [reply recipes/boost/all/test_package/CMakeLists.txt]: That's why I was hesitating and asking before. Imported targets give better error checking, but demand a newer cmake version. It's a trade off. I can also delete the commits again if you prefer.

Comment (uilianries) [reply recipes/boost/all/test_package/CMakeLists.txt]: I think you can decrease to Cmake 3.1 at least. Targets are available on that version. Thanks

Comment (mbodmer) [reply recipes/boost/all/test_package/CMakeLists.txt]: OK, I was only referring to the cmake documentation. will do that

**Comment by mbodmer** [comment]
> Now we have a consensus haha!

Great! :-) I wouldn't have minded to revert however. I am emotionless on this :-)

**Comment by uilianries** [comment]
@SSE4 You have some conflicts now :(

**Comment by danimtb** [comment]
Sorry for this. Please @mbodmer solve conflicts here. Thanks!

**Comment by mbodmer** [comment]
Ok, fixed it. Not much left however. Also ok if you don't merge it.

**Comment by danimtb** [comment]
Still some conflicts here. Could be related to merging another PR

**Pull request closed without merging**